client/jetpack-onboarding/main.jsx

```diff
@@ -141,6 +142,16 @@ class JetpackOnboardingMain extends React.PureComponent {
 				<DocumentHead
 					title={ get( STEP_TITLES, stepName ) + ' ‹ ' + translate( 'Jetpack Start' ) }
 				/>
+				{ /* It is important to use `<QuerySites siteId={ siteSlug } />` here, however wrong that seems.
+				   * The reason is that we rely on an `isRequestingSite()` check to tell whether we've
+				   * finished fetching site details, which will tell us whether the site is connected,
+				   * which we need in turn to conditionally send JPO auth credentials (see below).
+				   * However, if we're logged out, we cannot `<QuerySites allSites />`,
+				   * since the concept of "all of a user's sites" doesn't make sense if there is no user.
+				   * We also cannot `<QuerySites siteId={ siteId } />` prior to having obtained the `siteId`
+				   * through JPO -- a Catch-22 situation.
+				   * Fortunately, querying by `siteSlug` works, since it's supported by underlying actions. */ }
+				<QuerySites siteId={ siteSlug } />
```

Comment:
The downside with this is that we'll always get a 404 for unconnected sites:

![](https://cldup.com/Kp_mMq30On.png)

Reply:
Good spot 🙁 

Not sure we can do a lot about this since it's part of our "bootstrapping" where we're trying to find out whether the site is connected -- so arguably, this is relevant information, since we find out that it isn't...